### PR TITLE
Add tmp directory for temp output

### LINKED ISSUE #3904: Update FASTK_FASTK module to store and cleanup tmp files locally.
### Have you checked the docs?

- [X] [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
- [X] [nf-core modules documentation](https://nf-co.re/docs/contributing/modules)

### Description of the bug

Intermediate .fastk files are being currently and not being cleared up. The fix will be to add a local directory for these that can be deleted and the end of the process.

### Command used and terminal output

_No response_

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by weaglesBio**
## PR checklist

Closes #3904 

- [X] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

Intermediate .fastk files are being created currently to /tmp and not being cleared up. A local directory (tmp) was created and the intermediate outputs written to there (using the -P on the FASTK method), so these are now written to the work directory and can be cleared at the end of the process.

### CODE REVIEW

**Review by mahesh-panchal — COMMENTED** [review]
I'm not sure what problem this is solving.
There's nothing in the `tmp` folder at the end of the run (at least with test data).

File: modules/nf-core/fastk/fastk/main.nf [hunk 1]
```diff
@@ -26,11 +26,13 @@ process FASTK_FASTK {
     def prefix = task.ext.prefix ?: "${meta.id}"
     def FASTK_VERSION = 'f18a4e6d2207539f7b84461daebc54530a9559b0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
     """
+    mkdir tmp
     FastK \\
         $args \\
         -T$task.cpus \\
         -M${task.memory.toGiga()} \\
         -N${prefix}_fk \\
+        -Ptmp \\
```
Comment (FriederikeHanssen) [inline modules/nf-core/fastk/fastk/main.nf:35]: is it right that there is no space? Does just `.` work, so nextflow assigns the proper temp directory?

Comment (mahesh-panchal) [reply modules/nf-core/fastk/fastk/main.nf:35]: The original bug report was mentioning there were hidden files cluttering up the work directory, but I'm not sure specifically what these are. The tool creates some hidden files that are specifically needed for the output channels and these remain in the current directory and not in `tmp`. That folder just stays empty.

**Comment by muffato** [comment]
In my runs too, FastK does clean-up `/tmp`. I think this change is to improve the handling of interrupted runs of FastK, for instance on a HPC if the job is killed due to exceeding the memory usage. Instead of leaving temporary files in `/tmp`, the files would now be under the process' work directory itself (in the `tmp` sub-directory), which is easier to manage / clean up.

@weaglesBio . Can you tell us more ?

**Comment by muffato** [comment]
Another question, @weaglesBio : have you tried adding `-Ptmp` through `ext.args` ? With some luck it just works without having to pre-create the directory, meaning that you could implement the `-P` option in your pipeline rather than changing the module for everyone. Actually, `-P.` may work too ?

**Comment by SPPearce** [comment]
Is this issue still a problem for you @weaglesBio ?

**Comment by mahesh-panchal** [comment]
I think Matthieu's solution is the answer and this doesn't need to be added. By using `-P.` in ext.args means that it can be cleaned up with `nextflow clean` too after the workflow has successfully completed ( although I don't know how much of an issue it is to keep the work directory around until the other completes )

**Comment by SPPearce** [comment]
Closed for now following discussions.

**Pull request closed without merging**